components/automate-ui/src/app/entities/layout/layout.facade.ts

```diff
@@ -40,35 +49,64 @@ export class LayoutFacadeService {
       (notifications: Notification[]) => {
         this.showLicenseNotification =
           notifications &&  notifications.some(n => n.type === Type.license);
-        this.updateContentHeight(
-          this.showLicenseNotification ? Height.Banner : Height.Navigation);
+          if (this.showLicenseNotification) {
+            this.totalElementsHeights += Height.Banner;
+            this.updateContentHeight();
+          } else {
+            this.totalElementsHeights -= Height.Banner;
+            this.updateContentHeight();
+          }
+      });
+
+    this.store.select(allProjects).pipe(
+        takeUntil(this.isDestroyed)
+      ).subscribe((projectList: Project[]) => {
+        if(projectList.some(p => p.status === 'EDITS_PENDING')) {
+          this.totalElementsHeights += Height.PendingEditsBar;
+          this.updateContentHeight();
+        }
       });
+      this.totalElementsHeights += Height.Navigation;
+      this.updateContentHeight();
+  }
+
+  ngOnDestroy(): void {
+    this.isDestroyed.next(true);
+    this.isDestroyed.complete();
+  }
+
+  /* 
+    Pass true for fullPage
+  */
+  updateContentHeight(fullPage? : boolean): void {
+    const combinedHeights = fullPage ? 0 : this.totalElementsHeights;
+    this.contentHeight = `calc(100% - ${combinedHeights}px))`;
   }
 
   hasGlobalNotifications(): boolean {
-    return this.headerHeight === Height.Banner;
+    return this.showLicenseNotification;
   }
 
   ShowPageLoading(showLoading: boolean): void {
     this.store.dispatch(new ShowPageLoading(showLoading));
   }
 
   showFullPage(): void {
-    this.contentHeight = '100%';
+    this.updateContentHeight(true);
     this.showSidebar = false;
     this.showHeader = false;
   }
 
   hideFullPage(): void {
-    this.updateContentHeight(this.headerHeight);
+    this.updateContentHeight();
     this.showSidebar = true;
     this.showHeader = true;
   }
 
-  private updateContentHeight(height: string): void {
-    this.headerHeight = height;
-    this.contentHeight = `calc(100% - ${this.headerHeight})`;
-  }
+  // private updateContentHeight(height: string): void {
+  //   this.headerHeight = height;
+  //   this.contentHeight = `calc(100% - ${this.headerHeight})`;
+  // }
```

Comment:
Can we drop this code altogether?

Reply:
@tylercloke yes, i will. I left a note in the PR that I'm not finished with the layout height adjustments for the bottom user notification bars yet.